startupscript/butane/nvidia-runtime.yaml

```diff
@@ -0,0 +1,11 @@
+# Name: nvidia-runtime.yaml
+# Description: This snippet is used to reserve all GPUs on the host machine for the NVIDIA runtime.
+# Start of snippet from startupscript/butane/nvidia-runtime.yaml
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - driver: nvidia
+              count: all
+              capabilities: [gpu]
```

Comment:
After trying a bunch of different things, I couldn't make a static docker compose work, which is really unfortunate. But at least this block here is basically all we need to tell docker that the container needs access to nvidia gpu devices on the host, so in a way, this is better than having to paste some GPU block into every devcontainer that wants to be able to use gpus.

Reply:
docker compose up failure, I tried various values here too 😔